x/wasm/keeper/keeper_test.go

```diff
@@ -2188,3 +2188,81 @@ func TestCoinBurnerPruneBalances(t *testing.T) {
 		})
 	}
 }
+
+func TestIteratorAllContract(t *testing.T) {
+	ctx, keepers := CreateTestInput(t, false, AvailableCapabilities)
+	example1 := InstantiateHackatomExampleContract(t, ctx, keepers)
+	example2 := InstantiateHackatomExampleContract(t, ctx, keepers)
+	example3 := InstantiateHackatomExampleContract(t, ctx, keepers)
+	example4 := InstantiateHackatomExampleContract(t, ctx, keepers)
+
+	var allContract []string
+	keepers.WasmKeeper.IterateContractInfo(ctx, func(addr sdk.AccAddress, _ types.ContractInfo) bool {
+		allContract = append(allContract, addr.String())
+		return false
+	})
+	require.Contains(t, allContract, example1.Contract.String())
+	require.Contains(t, allContract, example2.Contract.String())
+	require.Contains(t, allContract, example3.Contract.String())
+	require.Contains(t, allContract, example4.Contract.String())
+}
```

Comment:
Thanks for adding this test. You confirm that you run through all instances.
For bonus points you would check for non duplicates and order. 
I would probably simply compare `expContracts := []string{example1.Contract.String(), example2.Contract.String()...}` against the `gotContracts` list